oak-segment-azure/src/test/java/org/apache/jackrabbit/oak/segment/azure/AzureSegmentArchiveWriterTest.java

```diff
@@ -86,15 +88,17 @@ public void retryWhenFailureOnWriteBinaryReferences_eventuallySucceed() throws E
                 .when(writeBinaryReferencesRequest, Times.once())
                 .respond(response().withStatusCode(201));
 
+        SegmentArchiveWriter writer = createSegmentArchiveWriter();
```

Comment:
in azure sdk 12 the retryOptions are set when doing the connection to azure https://github.com/ierandra/jackrabbit-oak/blob/931cc3bae6245df686d2206586dfda32f3194f78/oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/AzurePersistenceManager.java#L195
Once the connection is done the options cannot be changed. 
So when doing the testing all the mocking must be done before the connection. Because of this reason I change the test to reflect this.